implementation/configuration/src/configuration_impl.cpp

```diff
@@ -1380,6 +1382,64 @@ void configuration_impl::load_trace_filter_match(
     }
 }
 
+int configuration_impl::get_nic(const std::string& _local_address)
+{
+    // creating structures to browse through the available network interfaces
+    struct ifaddrs* ifaddr, *ifa;
+    int family, n;
+    sockaddr_in6* p_addrIP6;
+    std::vector<int> nic(0);    // the network interface card id. Vector in case 
+                    // more than one interface contains the searched IP address
+    struct sockaddr_in6 searched_address;
+    searched_address.sin6_family = AF_INET6;
+    searched_address.sin6_addr = in6addr_any;
+    searched_address.sin6_port = htons( 0xdead );
+    inet_pton( AF_INET6, _local_address.c_str(), &searched_address.sin6_addr ); // actually copying 
+
+    if (getifaddrs(&ifaddr) == -1) // "-1" is an error value
+    {
+        VSOMEIP_ERROR << __func__ <<  
+            strerror(errno) << ::std::endl;
+        return -1;
+    }
+
+    /* Walk through linked list of network interfaces*/
+    for (ifa = ifaddr, n = 0; ifa != NULL; ifa = ifa->ifa_next, n++)
+    {
+        if (ifa->ifa_addr == NULL)
+            continue;
+        family = ifa->ifa_addr->sa_family;
+
+        /* For an AF_INET6 interface address, compare the address of the NIC
+        with the searched address */
+        if (family == AF_INET6)
+        {
+            p_addrIP6 = reinterpret_cast<sockaddr_in6*>(ifa->ifa_addr);
+            if (::bcmp(p_addrIP6->sin6_addr.s6_addr, searched_address.sin6_addr.s6_addr, 16) == 0)
+            {
+                nic.push_back(if_nametoindex(ifa->ifa_name));
+            }
+        }
+    }
+    if (nic.size() > 1)
+    {
+        VSOMEIP_ERROR << __func__ << " found multiple network interfaces " <<
```

Comment:
Maybe start the log message with :
<< __func__ << ": Found....
And again now sure if using ERROR is the best approach